### PR TITLE
Sscs 7937 joint party upload response

### PR DESCRIPTION
**Pull request by michaellavelle**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-7937

### Change description ###

* Adding joint party dwpUploadResponse notifications for hearing attended (oral)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java [hunk 1]
```diff
@@ -159,11 +161,13 @@ public void shouldGiveCorrectDocmosisIdForDirectionIssued(DirectionType directio
...
                 new Object[]{"90f0ed29-a616-4ce0-b4ef-108391f5d90e", Collections.EMPTY_LIST, null, null, ONLINE, getTemplateName(DWP_RESPONSE_RECEIVED_NOTIFICATION), null},
+                new Object[]{"d994236b-d7c4-44ef-9627-12372bb0434a", Arrays.asList("7d36718b-1193-4b3d-86bd-db54612c5363"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, JOINT_PARTY), null},
                 new Object[]{"a3b22e07-e90b-4b52-a293-30823802c209", Arrays.asList("aaa1aad4-7abc-4a7a-b8fb-8b0567c09365"), null, null, PAPER, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPELLANT), null},
-                new Object[]{"b9e47ec4-3b58-4b8d-9304-f77ac27fb7f2", Arrays.asList("e3f71440-d1ac-43c8-a8cc-a088c4f3c959"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPELLANT), null},
+                new Object[]{"d994236b-d7c4-44ef-9627-12372bb0434a", Arrays.asList("7d36718b-1193-4b3d-86bd-db54612c5363"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPELLANT), null},
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java:168]: Has Evidence reminder templates changed?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java [hunk 2]
```diff
@@ -159,11 +161,13 @@ public void shouldGiveCorrectDocmosisIdForDirectionIssued(DirectionType directio
...
                 new Object[]{"a3b22e07-e90b-4b52-a293-30823802c209", Arrays.asList("aaa1aad4-7abc-4a7a-b8fb-8b0567c09365"), null, null, PAPER, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPOINTEE), null},
-                new Object[]{"b9e47ec4-3b58-4b8d-9304-f77ac27fb7f2", Arrays.asList("e3f71440-d1ac-43c8-a8cc-a088c4f3c959"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPOINTEE), null},
+                new Object[]{"d994236b-d7c4-44ef-9627-12372bb0434a", Arrays.asList("7d36718b-1193-4b3d-86bd-db54612c5363"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, APPOINTEE), null},
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java:170]: Is this correct?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java [hunk 3]
```diff
@@ -142,10 +142,12 @@ public void shouldGiveCorrectDocmosisIdForDirectionIssued(DirectionType directio
     @SuppressWarnings({"Indentation", "unused"})
     private Object[] templateIdsWithHearingAndEventTypesWelsh() {
         return new Object[]{
+                new Object[]{"738b40f4-e7fe-4121-b09d-0a76be4d2fe9", Arrays.asList("00c36e89-6cf0-42a1-b080-6eac4edfca25", "7d36718b-1193-4b3d-86bd-db54612c5363"), null, null, ORAL, getTemplateName(EVIDENCE_REMINDER_NOTIFICATION, JOINT_PARTY), null},
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/config/NotificationConfigTest.java:145]: Has Evidence Reminder config been added?

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/tya/NotificationsIt.java [hunk 4]
```diff
@@ -567,7 +567,43 @@ private void validateLetterNotifications(List<String> expectedLetterTemplateIds,
                         "0",
                         "0",
                         "2"
-                }
+                },
+                new Object[]{
+                        EVIDENCE_REMINDER_NOTIFICATION,
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/tya/NotificationsIt.java:572]: Better not to add this

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/tya/NotificationsIt.java [hunk 5]
```diff
@@ -2436,8 +2472,8 @@ private void validateLetterNotifications(List<String> expectedLetterTemplateIds,
             new Object[]{
                 EVIDENCE_REMINDER_NOTIFICATION,
                 "oral",
-                Arrays.asList("b9e47ec4-3b58-4b8d-9304-f77ac27fb7f2"),
-                Arrays.asList("e3f71440-d1ac-43c8-a8cc-a088c4f3c959"),
+                Arrays.asList("d994236b-d7c4-44ef-9627-12372bb0434a"),
+                Arrays.asList("7d36718b-1193-4b3d-86bd-db54612c5363"),
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/tya/NotificationsIt.java:2476]: Has this changed?

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/JointPartyFunctionalTest.java [hunk 6]
```diff
@@ -1,26 +1,40 @@
...
+    private static final String ORAL = "oral";
     @Value("${notification.english.appealLapsed.joint_party.emailId}")
     private String appealLapsedJointPartyEmailId;
     @Value("${notification.english.appealLapsed.joint_party.smsId}")
     private String appealLapsedJointPartySmsId;
-
+    @Value("${notification.english.oral.evidenceReminder.joint_party.emailId}")
+    private String oralEvidenceReminderJointPartyEmailId;
+    @Value("${notification.english.oral.evidenceReminder.joint_party.smsId}")
+    private String oralEvidenceReminderJointPartySmsId;
```
Comment (ryandac) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/JointPartyFunctionalTest.java:33]: remove evidence Reminder

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/JointPartyFunctionalTest.java [hunk 7]
```diff
@@ -72,9 +94,14 @@ private String getFieldValue(NotificationEventType notificationEventType, String
...
-        int expectedNumberOfLettersIsTwo = 2;
+        final int expectedNumberOfLettersIsTwo = 2;
+        final int expectedNumberOfLettersIsZero = 0;
+        final boolean isDocmosisLetterTrue = true;
+        final boolean isDocmosisLetterFalse = false;
         return new Object[]{
-            new Object[]{APPEAL_LAPSED_NOTIFICATION, expectedNumberOfLettersIsTwo}
+            new Object[]{APPEAL_LAPSED_NOTIFICATION, NO_HEARING_TYPE, expectedNumberOfLettersIsTwo, isDocmosisLetterFalse},
+            new Object[]{DWP_UPLOAD_RESPONSE_NOTIFICATION, ORAL, expectedNumberOfLettersIsTwo, isDocmosisLetterTrue},
+            new Object[]{EVIDENCE_REMINDER_NOTIFICATION, ORAL, expectedNumberOfLettersIsZero, isDocmosisLetterFalse}
```
Comment (ryandac) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/sya/notifications/JointPartyFunctionalTest.java:104]: Remove evidence reminder

File: src/e2e/resources/jointParty/oral-evidenceReminderCallback.json [hunk 8]
```diff
@@ -0,0 +1,717 @@
+{
```
Comment (ryandac) [inline src/e2e/resources/jointParty/oral-evidenceReminderCallback.json:1]: remove evidence reminder

File: src/main/resources/application.yaml [hunk 9]
```diff
@@ -431,24 +431,30 @@ notification:
...
         appellant:
-          emailId: b9e47ec4-3b58-4b8d-9304-f77ac27fb7f2
-          smsId: e3f71440-d1ac-43c8-a8cc-a088c4f3c959
+          emailId: d994236b-d7c4-44ef-9627-12372bb0434a
+          smsId: 7d36718b-1193-4b3d-86bd-db54612c5363
         appointee:
-          emailId: b9e47ec4-3b58-4b8d-9304-f77ac27fb7f2
-          smsId: e3f71440-d1ac-43c8-a8cc-a088c4f3c959
+          emailId: d994236b-d7c4-44ef-9627-12372bb0434a
+          smsId: 7d36718b-1193-4b3d-86bd-db54612c5363
```
Comment (ryandac) [inline src/main/resources/application.yaml:451]: remove evidence reminder changes

File: src/main/resources/application.yaml [hunk 10]
```diff
@@ -798,6 +807,9 @@ notification:
         representative:
           emailId: 3260b6ec-9930-44b6-aa65-58905dd9a5b3
           smsId: a2745961-214a-4138-95b0-89300b395a3e
+        joint_party:
+          emailId: 738b40f4-e7fe-4121-b09d-0a76be4d2fe9
+          smsId: 00c36e89-6cf0-42a1-b080-6eac4edfca25
```
Comment (ryandac) [inline src/main/resources/application.yaml:812]: remove evidence reminder changes

File: src/test/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapperTest.java [hunk 11]
```diff
@@ -190,8 +190,8 @@ public void givenSubscriptions_shouldGetSubscriptionTypeListWithAppointee(Notifi
     }
 
     @Test
-    @Parameters({"APPEAL_LAPSED_NOTIFICATION, paper", "APPEAL_LAPSED_NOTIFICATION, oral",})
-    public void givenSubscriptions_shouldGetSubscriptionTypeListWithAppointeeAndKointParty(NotificationEventType notificationEventType, String hearingType) {
+    @Parameters({"APPEAL_LAPSED_NOTIFICATION, paper", "APPEAL_LAPSED_NOTIFICATION, oral", "EVIDENCE_REMINDER_NOTIFICATION, oral", "DWP_UPLOAD_RESPONSE_NOTIFICATION, oral"})
```
Comment (ryandac) [inline src/test/java/uk/gov/hmcts/reform/sscs/factory/CcdNotificationWrapperTest.java:193]: remove EVIDENCE_REMINDER_NOTIFICATION, "oral test

**Review by ryandac — CHANGES_REQUESTED** [review]
Evidence reminder changes have somehow made it into your PR, please remove them

**Review by ryandac — APPROVED** [review]

**Comment by michaellavelle** [comment]
Superseded by https://github.com/hmcts/sscs-track-your-appeal-notifications/pull/1212

**Pull request closed without merging**